lib/os/reboot.c

```diff
@@ -4,13 +4,15 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <reboot.h>
 #include <kernel.h>
 #include <sys/printk.h>
+#include <toolchain.h>
 
 extern void sys_arch_reboot(int type);
 extern void sys_clock_disable(void);
 
-void sys_reboot(int type)
+void FUNC_NORETURN sys_reboot(int type)
```

Comment:
While not wrong, should follow the project-wide convention. Also you used the following convention in the function definition in the header, so it would be good to do the same for the declaration.

```suggestion
FUNC_NORETURN void sys_reboot(int type)
```